arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -411,8 +411,8 @@ sealed class IO<out A> : IOOf<A> {
      * @param cb an asynchronous computation that might fail.
      * @see async for wrapping impure APIs without cancellation
      */
-    fun <E, A> cancellable(cb: ((IOResult<E, A>) -> Unit) -> IOOf<E, Unit>): IO<E, A> =
-      Async(true) { conn: IOConnection, cbb: (IOResult<E, A>) -> Unit ->
+    fun <A> cancellable(cb: ((Either<Throwable, A>) -> Unit) -> CancelToken<ForIO>): IO<A> =
+      Async { conn: IOConnection, cbb: (Either<Throwable, A>) -> Unit ->
```

Comment:
@colomboe Sorry i unconsciously removed this wile merging master in, would you mind adding it back?

Also cc @nomisRev this seems to be the cause of the stack overflows in this PR, but I'm unaware of the consequences of adding it back or why is it needed in the first place 🤔

Reply:
There was an incorrect overload when the change from `cancelable` to `cancellable` was made, which resulted in an infinite loop.

Might be related to that if this was introduced by a revert of `IO`.

Reply:
I've restored the change